crates/noirc_evaluator/src/ssa/code_gen.rs

```diff
@@ -948,4 +954,56 @@ impl<'a> IRGenerator<'a> {
         }
         last_value
     }
+
+    fn handle_if_expr(
+        &mut self,
+        env: &mut Environment,
+        cond: &ExprId,
+        cons: &ExprId,
+        alternative: Option<ExprId>,
+    ) -> Result<Value, RuntimeError> {
+        //jump instruction
+        let entry_block = self.context.current_block;
+        let condition = self
+            .expression_to_object(env, cond)
+            .map_err(|err| err.remove_span())
+            .unwrap()
+            .unwrap_id();
+        let jump_op = Operation::Jeq(condition, block::BlockId::dummy());
+        let jump_ins = self.context.new_instruction(jump_op, ObjectType::NotAnObject);
+
+        //Then block
+        block::new_sealed_block(&mut self.context, block::BlockType::Normal, true);
+        //Parse statements of THEN branch
+        self.expression_to_object(env, cons).map_err(|err| err.remove_span()).unwrap();
```

Comment:
Writing this for future reference: we'll need to save this value and create a PHI of the 'then' and 'else' branch values for the return value of an if expression eventually.

Reply:
yes